server/supernova/server/server.cpp

```diff
@@ -319,9 +319,17 @@ void realtime_engine_functor::init_thread(void) {
     name_current_thread(0);
 }
 
-void realtime_engine_functor::log_(const char* str) { instance->log_printf(str); }
+void realtime_engine_functor::log_(const char* str) {
+    // Silently drop messages; if instance is null it means something failed while constructing nova_server
+    if (instance)
+        instance->log_printf(str);
+}
 
 void realtime_engine_functor::log_printf_(const char* fmt, ...) {
+    // Silently drop messages; if instance is null it means something failed while constructing nova_server
+    if (instance == nullptr)
+        return;
+
```

Comment:
my thinking here was that any messages logged here are probably spurious and would distract from the actual error. this can only happen if we are in the middle of constructing the base classes of `nova_server` and something throws, so i'm relying on that thrown exception to be more informative